Update Cache component doc for 4.2 (#11477)
symfony/symfony-docs: cache.rst

                    ],
                    'acme.cache' => [
                        'adapter' => 'cache.adapter.memcached',
                    ],
                    'foobar.cache' => [
                        'adapter' => 'cache.adapter.memcached',
                        'provider' => 'memcached://user:password@example.com',
                    ],
                    'short_cache' => [
                        'adapter' => 'foobar.cache',
                        'default_lifetime' => 60,
                    ],
                ],
            ],
        ]);

Each pool manages a set of independent cache keys: keys of different pools
*never* collide, even if they share the same backend. This is achieved by prefixing
keys with a namespace that's generated by hashing the name of the pool, the name
of the compiled container class and a :ref:`configurable seed<reference-cache-prefix-seed>`
that defaults to the project directory.

Each custom pool becomes a service where the service id is the name of the pool
(e.g. ``custom_thing.cache``). An autowiring alias is also created for each pool
using the camel case version of its name - e.g. ``custom_thing.cache`` can be

> injected automatically by naming the argument ``$forecastCache`` and type-hinting it

Author: $customThingCache

The ``cache.acme`` pool is using the Memcached server on localhost and ``cache.foobar``
is using the Memcached server at example.com.

Each pool will manage a set of independent cache keys: keys of different pools
never collide even if they share the same backend. This is achieved by prefixing
keys with a namespace. This namespace is generated by hashing the name of the

> pool, the name of the compiled container class and a configurable seed that XXX HOWTO LINK TO reference/configuration/framework.rst#prefix_seed? XXX

Author: RST expert needed :)

Reviewer: Will take care of this tomorrow 👌🏻

Reviewer: Add this to `reference/configuration.framework.rst` file:

```diff
+ .. _reference-cache-prefix-seed

prefix_seed
...........

**type**: ``string`` **default**: ``null``

If defined, this value is used as part of the "namespace" generated for the
cache item keys. A common practice is to use the unique name of the application
(e.g. ``symfony.com``) because that prevents naming collisions when deploying
multiple applications into the same path (on different servers) that share the
same cache backend.

It's also useful when using `blue/green deployment`_ strategies and more
generally, when you need to abstract out the actual deployment directory (for
example, when warming caches offline).

```

And link it like this:
```suggestion
pool, the name of the compiled container class and a :ref:`configurable seed <reference-cache-prefix-seed>`
```
